drivers/spi/spi_dw.c

```diff
@@ -556,10 +553,6 @@ struct spi_dw_data spi_dw_data_port_0 = {
 const struct spi_dw_config spi_dw_config_0 = {
 	.regs = DT_INST_REG_ADDR(0),
 	.clock_frequency = INST_0_SNPS_DESIGNWARE_SPI_CLOCK_FREQ,
-#ifdef CONFIG_SPI_DW_PORT_0_CLOCK_GATE
```

Comment:
Could you use DTS settings instead?
Many drivers do, afaik. 
Like drivers/i2c/i2c_rv32m1_lpi2c.c
or drivers/i2c/i2c_mcux_lpi2c.c 

Would this require adding a specific property into binding file?

Reply:
It would require knowing the clocking info in DTS for an SoC we wanted to support.  Thus I felt it was easier to re-add this when we need that functionality.  (I hope by than we've also progressed clock control support in general in Zephyr).

Reply:
> It would require knowing the clocking info in DTS for an SoC we wanted to support. 

Yes, and don't we have the means already to know that info in DTS already? And if it's not present, then it will know there no such clock. I am really not keen into removing that support just yet, moreover that it's being properly configured on other driver.

Reply:
clock params in dts vary soc to soc right now.  So extracting the clock name & subsys is SoC specific.  Thus my comment about needing an example to make such a conversion.

Reply:
The dts attribute "clocks" seems generic to me: it's in dts/bindings/base/base.yaml

and that's exactly the attribute that may be used to set the info required here.

Reply:
So we can use `clocks` to get the device name.  That is easy enough.  Its the replacing CONFIG_SPI_DW_PORT_n_CLOCK_GATE_SUBSYS value that we can't get w/o a defined clock binding for a SoC that defines such a thing.

Reply:
clocks is an array, you can get the subsystem as well afaik, see for instance dts/arm/nxp/nxp_k6x.dtsi

Reply:
My point is the clock array is specific to each clock provider.  So w/o an actual dts / binding that we want to support I'm just putting random code into the tree that we have no way of even build testing.

I appreciate wanting to keep the functionality, but we need either some in tree user or I'd ask we drop it until we have one.  We can always revert the change and fix it up at the point we have an in tree user.